### DISCUSSION TITLE
Add an Event hook for after Gateway response is returned

### DISCUSSION THREAD

**Discussion by mylesthedev** [craftcms/commerce#2928] [Ideas]
There is already a hook `EVENT_BEFORE_GATEWAY_REQUEST_SEND` however there isn't one after the response is returned from the gateway.

The reason this would be helpful is some gateways return encoded responses and might need decoding (See https://github.com/PlatoCreative/commerce-payment-express/issues/5#issuecomment-1208879186) 

I believe this should happen after this line `$this->sendRequest($request);`

**Comment by anchovy** [comment]
We've got the same need. 

+1 for a `EVENT_AFTER_GATEWAY_REQUEST_RECIVED` - we need to capture a service fee added by the gateway provider (https://github.com/PlatoCreative/commerce-payment-express/issues/3)


PHP: 8.0.8
Craft: Craft Pro 3.7.50
Commerce: 3.4.16
Payment Express for Craft Commerce 2: 1.5.0

**Comment by TheCohorts-reuben** [comment]
Same here, this would be great to have!
